core/src/test/java/com/rudderstack/core/internal/ConfigDownloadServiceImplTest.kt

```diff
@@ -97,7 +103,57 @@ abstract class ConfigDownloadServiceImplTest {
 
         Awaitility.await().atMost(2, TimeUnit.SECONDS).untilTrue(isComplete)
     }
-
+    @Test
+    fun `test listener is fired when download is called`(){
+        val mockListener = mock<ConfigDownloadService.Listener>()
+        val isComplete = AtomicBoolean(false)
+        configDownloadServiceImpl.addListener(mockListener, 0)
+        configDownloadServiceImpl.download(callback = { success, rudderServerConfig, lastErrorMsg ->
+            isComplete.set(true)
+        })
+        while (!isComplete.get()){}
+        verify(mockListener, times(1)).onDownloaded(true)
+    }
+    @Test
+    fun `test listener is fired when download replayed`(){
+        val mockListener = mock<ConfigDownloadService.Listener>()
+        val isComplete = AtomicBoolean(false)
+        configDownloadServiceImpl.download(callback = { success, rudderServerConfig, lastErrorMsg ->
+            isComplete.set(true)
+        })
+        while (!isComplete.get()){}
+        isComplete.set(false)
+        dummyWebService.nextStatusCode = 400
+        configDownloadServiceImpl.download(callback = { success, rudderServerConfig, lastErrorMsg ->
+            isComplete.set(true)
+        })
+        while (!isComplete.get()){}
+        configDownloadServiceImpl.addListener(mockListener, 1)
+        verify(mockListener, times(1)).onDownloaded(false)
+    }
```

Comment:
This test case takes longer to execute. Is it possible to simplify this further?
Refer here for timings: 
<img width="607" alt="image" src="https://github.com/rudderlabs/rudder-sdk-android/assets/64667840/e41327a1-faa5-43cd-a4d4-566af19718ca">

IMO this is not good and should be improved.